main/src/ca/uwaterloo/flix/language/phase/unification/set/SetFormula.scala

```diff
@@ -652,4 +651,235 @@ object SetFormula {
   def mkDifference(f1: SetFormula, f2: SetFormula): SetFormula =
     mkInter(f1, mkCompl(f2))
 
+  //
+  // Other Functions
+  //
+
+  /**
+    * Propagation uses the following rewrites:
+    *   - `x ∩ f == x ∩ f[x -> Univ]`
+    *   - `!x ∩ f == !x ∩ f[x -> Empty]`
+    *   - `x ∪ f == x ∪ f[x -> Empty]`
+    *   - `!x ∪ f == !x ∪ f[x -> Univ]`
+    *
```

Comment:
How many of these did we check in Isabelle? What did Edward think?

Reply:
All four - and proven for all formulas `x`, not just cst,elems,vars